### PR TITLE
ROX-13261: DryRunUpdate on collection datastore

### PR DESCRIPTION
**Pull request by keyallis**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: central/resourcecollection/datastore/datastore_impl.go [hunk 1]
```diff
@@ -348,8 +348,14 @@ func (ds *datastoreImpl) updateCollectionWorkflow(ctx context.Context, collectio
...
-	ds.lock.Lock()
-	defer ds.lock.Unlock()
+	// if this a dryrun we don't ever end up calling upsert so we only need to get a read lock
+	if dryrun {
+		ds.lock.RLock()
+		defer ds.lock.RUnlock()
+	} else {
+		ds.lock.Lock()
+		defer ds.lock.Unlock()
+	}
```
[the remark this answers is not in the document]
Comment (keyallis) [reply central/resourcecollection/datastore/datastore_impl.go:358]: I don't think I understand what you mean by this. From what I can logically reason it seems like that would not be an issue and we only need to acquire a read lock for the update dryrun operation

[the remark this answers is not in the document]
Comment (md2119) [reply central/resourcecollection/datastore/datastore_impl.go:358]: Fair enough. I expanded the diff to check how the `names` is used.

**Review by md2119 — APPROVED** [review]

**Pull request merged**